【feature】add registry and discovery ablity for springboot

justforstudy-A opened the pull request:
What type of PR is this?

/feat

Which issue(s) this PR fixes:

Fixes #814

Use case description: make springboot application support registry and discovery

Does this PR introduce a user-facing change?

No

Additional documentation e.g., usage docs, etc.:

No

provenceee committed:
*add zone router

justforstudy-A committed:
springboot注册能力提交

修复依赖

修复UT

修改根路径

优化代码

修改默认缓存时间

增加zk状态检测; 修正review意见

修正ut

luanwenfei-venus commented on sermant-plugins/sermant-springboot-registry/springboot-registry-plugin/src/main/java/com/huawei/discovery/retry/Retry.java:
```diff
@@ -0,0 +1,101 @@
+ */
+public interface Retry {
+    /**
+     * 创建重拾器
```
重拾->重试

justforstudy-A replied:
已修改

luanwenfei-venus commented on sermant-plugins/sermant-springboot-registry/springboot-registry-plugin/src/main/java/com/huawei/discovery/retry/Retry.java:
```diff
@@ -0,0 +1,101 @@
+     * 配置名
+     *
+     * @return 重试器名称
+     */
+    String name();
+
```
这里是一个方法 获取配置名？

justforstudy-A replied:
仅仅是配置名， 后续自定义配置会用到， 区分重试配置， 目前没用到

luanwenfei-venus commented on sermant-plugins/sermant-springboot-registry/springboot-registry-plugin/src/main/java/com/huawei/discovery/retry/RetryConfig.java:
```diff
@@ -0,0 +1,148 @@
+ * @since 2022-09-28
+ */
+public class RetryConfig {
+    private static final int DEFAULT_MAX_RETRY = 3;
+
+    private static final long DEFAULT_WAIT_RETRY_MS = 1000L;
```
需要放到配置文件吗

justforstudy-A replied:
这个创建的时候会指定， 对应配置在LbConfig里

develpoerX committed:
调整checkstype格式问题、增加动态配置校验

lilai23 commented on sermant-plugins/sermant-springboot-registry/config/config.yaml:
```diff
@@ -0,0 +1,7 @@
+# consul发现配置
+sermant.springboot.registry:
+  realmName: gateway.t3go.com.cn
```
这个跟consul发现配置没关系，应该是被拦截域名；realmName建议改成domain；不要把真实域名放在开源仓库上

justforstudy-A replied:
已修改

lilai23 replied:
我的意思是建议realmName这个key改成domain或者domainName，参考DNS的全称

lilai23 commented on sermant-plugins/sermant-springboot-registry/springboot-registry-plugin/pom.xml:
```diff
@@ -0,0 +1,174 @@
+        <dependency>
+            <groupId>com.squareup.okhttp3</groupId>
+            <artifactId>okhttp</artifactId>
+            <version>4.1.0</version>
```
版本统一管理下

justforstudy-A replied:
已修改

lilai23 commented on sermant-plugins/sermant-springboot-registry/springboot-registry-plugin/src/main/java/com/huawei/discovery/config/DiscoveryPluginConfig.java:
```diff
@@ -0,0 +1,56 @@
+    /**
+     * 是否打印统计日志
+     */
+    private boolean loggerFlag = false;
```
配置文件里配置下默认值吧，这个是统计数量的，建议叫做enableRequestCount

justforstudy-A replied:
已修改

provenceee commented on sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/plugin/config/ServiceMeta.java:
```diff
@@ -61,6 +61,16 @@ public class ServiceMeta implements BaseConfig {
 
     private Map<String, String> parameters;
 
+    private String zone = "default";
```
默认值不要用"default"，可以去掉默认值或者改成""

justforstudy-A replied:
已修改

lilai23 commented on sermant-plugins/sermant-springboot-registry/springboot-registry-plugin/src/main/java/com/huawei/discovery/utils/RequestInterceptorUtils.java:
```diff
@@ -0,0 +1,214 @@
+    }
+
+    /**
+     * 解析url参数信息 http://gateway.com.cn/serviceName/sayHell?name=1
```
隐去真实域名

justforstudy-A replied:
已修改

xuezechao1 commented on sermant-plugins/sermant-springboot-registry/springboot-registry-plugin/src/main/java/com/huawei/discovery/entity/SimpleRequestRecorder.java:
```diff
@@ -0,0 +1,78 @@
+            LOGGER.info("SimpleRequestRecorder has over the max num of long, it has been reset to 0!");
+        }
+        if (discoveryPluginConfig.isLoggerFlag()) {
+            LOGGER.log(Level.INFO, String.format(Locale.ENGLISH,
```
建议日志打印的方式保持一致是不是好一点

justforstudy-A replied:
已修改

justforstudy-A committed:
调整缓存刷新机制

chengyouling committed:
Merge branch 'huaweicloud:develop' into springboot-registry

justforstudy-A committed:
调整缓存刷新机制

justforstudy-A committed:
Merge branch 'springboot-registry' of github.com:chengyouling/Sermant into springboot-registry

justforstudy-A committed:
checkstyle & review意见

修复构建url问题

checkstyle & review意见

修改日志级别， 修复黑白名单bug

重试仅连接异常

修复当zk连接延时时，无法注册的问题

修复UT

justforstudy-A committed:
Merge branch 'develop' into springboot-registry
